### ISSUE TITLE
Sheet dismisses itself when using navigation

### ISSUE THREAD

**Issue by Lightwood13** [CoolONEOfficial/NativePartialSheet#6]
Hello, thank you for you library. I've been using it in my project and noticed an issue. I'm using library [NavigationBackport](https://github.com/johnpatrickmorgan/NavigationBackport) to backport NavigationStack to iOS 15.

When using NavigationStack and sheets on 2 screens, sheet on the second screen dismisses itself immediately after it is opened.
Also, when opening sheet on the first screen, it shows warnings in logs:
```=== AttributeGraph: cycle detected through attribute 59088 ===
=== AttributeGraph: cycle detected through attribute 58936 ===
=== AttributeGraph: cycle detected through attribute 58936 ===
=== AttributeGraph: cycle detected through attribute 59088 ===
=== AttributeGraph: cycle detected through attribute 59088 ===
```

Minimal code example to reproduce:
```
import SwiftUI
import NavigationBackport
import NativePartialSheet

enum Route {
    case SecondScreen
}

@main
struct iOSApp: App {
    
    var body: some Scene {
        WindowGroup {
            NBNavigationStack {
                FirstScreen()
                    .nbNavigationDestination(for: Route.self) { route in
                        switch (route) {
                        case .SecondScreen:
                            SecondScreen()
                        }
                    }
            }
        }
    }
}

struct FirstScreen: View {
    @EnvironmentObject private var navigator: PathNavigator
    @State private var sheetPresented = false

    var body: some View {
        VStack {
            Button(action: { sheetPresented = true }) {
                Text("Show sheet")
            }
            .sheet(isPresented: $sheetPresented, content: {
                Text("Sheet 1")
            })
            
            Button(action: { navigator.push(Route.SecondScreen) }) {
                Text("Go to second screen")
            }
        }
    }
}

struct SecondScreen: View {
    @State private var sheetPresented = false
    
    var body: some View {
        Button(action: { sheetPresented = true }) {
            Text("Show sheet")
        }
        .sheet(isPresented: $sheetPresented) {
            Text("Sheet 2")
        }
    }
}
```

Video with bug:

https://github.com/CoolONEOfficial/NativePartialSheet/assets/33136089/36d1e06e-de13-459a-ad76-345f50ce3a33

I've also tested with navigation and sheets from SwiftUI, and second time without navigation and with your library and couldn't reproduce an issue in either case. So it seems like some kind of weird interaction between libraries is causing this bug.
Thank you in advance!

**Comment by CoolONEOfficial** [comment]
Hi, looks quite interesting 🤨 
I'll try to figure out how to fix that behavior if it's possible

**Comment by elkopass** [comment]
Oh, I have the same problem, I really hope that u can fix it!

**Comment by CoolONEOfficial** [comment]
Fixed in 3.0.0 release (#7)
